packages/@tinacms/schema-tools/src/schema/TinaSchema.ts

```diff
@@ -116,37 +116,41 @@ export class TinaSchema {
       format: collection.format || 'md',
     };
   };
+
   public getCollections = () => {
     return (
       this.schema.collections.map((collection) =>
         this.getCollection(collection.name)
       ) || []
     );
   };
+
   public getCollectionByFullPath = (filepath: string) => {
     const fileExtension = filepath.split('.').pop();
-    const normalizedPath = filepath.replace(/\\/g, '/');
+    const canonicalFilepath = canonicalPath(filepath);
 
     const possibleCollections = this.getCollections().filter((collection) => {
       // filter out file extensions that don't match the collection format
       if (
-        !normalizedPath.endsWith(`.gitkeep.${collection.format || 'md'}`) &&
+        !canonicalFilepath.endsWith(`.gitkeep.${collection.format || 'md'}`) &&
         fileExtension !== (collection.format || 'md')
       ) {
         return false;
       }
       if (collection?.match?.include || collection?.match?.exclude) {
         // if the collection has a match or exclude, we need to check if the file matches
         const matches = this.getMatches({ collection });
-        const match = picomatch.isMatch(normalizedPath, matches);
+        const match = picomatch.isMatch(canonicalFilepath, matches);
 
         if (!match) {
           return false;
         }
       }
-      // add / to the end of the path if it is not "''"
-      const path = collection.path ? collection.path.replace(/\/?$/, '/') : '';
-      return normalizedPath.startsWith(path);
+      const collectionPath = canonicalPath(collection.path);
+      return (
+        collectionPath == '' ||
+        canonicalFilepath.startsWith(`${collectionPath}/`)
+      );
```

Comment:
[nitpick] Use `===` over `==`

```suggestion
if (collectionPath === '') {
  return true;
}
return canonicalFilepath.startsWith(`${collectionPath}/`);
```